### DISCUSSION TITLE
problem to find another solutions

### DISCUSSION THREAD

**Discussion by gregy4** [google/or-tools#3880] [CP-SAT questions]
Hello,
  related to #3870. Even with given initial solution via hints, cp-sat solver is much slower compared to constraint solver. Unfortunatelly hours without a progress far away from proved optimal value (31 and 25) doesn't look promising. The solver is still capable to prove better solution found by constraint solver.

  I tried another data instance and results are similar.

![comparison_of_solvers](https://github.com/google/or-tools/assets/5640112/d72c0ba7-c98f-4932-ada1-3a942116be69)
![comparison_of_solvers2](https://github.com/google/or-tools/assets/5640112/f3b757ba-2fc0-4473-97d2-3e3f7746646c)

[model-comp.zip](https://github.com/google/or-tools/files/12248872/model-comp.zip)
[model-comp2.zip](https://github.com/google/or-tools/files/12248876/model-comp2.zip)

  Is there a chance to find better solutions ?  I though about playing with parameters and maybe branching strategy (I know very different to constraint solvers) to see some progress but in case main bottleneck is for example performance of sat solver it doesn't make sense.

Thanks,
Jan

**Comment by lperron** [comment]
Thanks again for the models. Can you explain the graph ? y axis is time ?

**Reply by gregy4** [reply]
Yes, both models minimize number of used workers. First model starts with 40 workers and second model with 50 workers. On x axis is number of workers and on y axis is number of elapsed seconds when a solution with given number of workers was found.

**Comment by lperron** [comment]
Thanks for the explanation. Indeed, we have no lns dedicated to remove all tasks from a resource. I need some time to implement this.
You can do it at the model level as a workaround. But definitely choco is doing a good job at it.

**Comment by lperron** [comment]
How do you define the cost ?

**Reply by gregy4** [reply]
As sum of used workers. There’s a variable for every worker to identify state “worker is used”. The variable is true when a worker is used by any operation (optional variable representing worker is used by an operation is active), it is done by implication. “Worker is not used” imply that all optional intervals that belongs to a worker are inactive.

**Comment by lperron** [comment]
OK. Are the workers equivalent ?

Laurent Perron | Operations Research | ***@***.*** | (33) 1 42 68 53
00

**Reply by gregy4** [reply]
Workers are divided into groups with some skill and an operation needs amount of workers with the skill. Number of possible workers for an operation can be further limited - like use only local workers and not workers available for all stations (an operation always belong to some station).

In case a worker can be used by an operation, it is equivalent to another possible worker.
